core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java

```diff
@@ -133,6 +133,30 @@ public class DateTimeFormatters {
           .toFormatter(Locale.ROOT)
           .withResolverStyle(ResolverStyle.STRICT);
 
+  // MMDD
+  public static final DateTimeFormatter DATE_FORMATTER_SINGLE_DIGIT_MONTH =
+      new DateTimeFormatterBuilder()
+          .parseDefaulting(YEAR, 2000)
+          .appendPattern("Mdd")
+          .toFormatter()
+          .withResolverStyle(ResolverStyle.STRICT);
+
+  // MMDD
+  public static final DateTimeFormatter DATE_FORMATTER_NO_YEAR =
+      new DateTimeFormatterBuilder()
+          .parseDefaulting(YEAR, 2000)
```

Comment:
There is a pattern with year below for format `YYYYMMDD`
```suggestion
```

Reply:
Similar to the comment above, this formatter will accept a 4 digit argument. For example `1015` is treated as `October 15, 2000`. The `parseDefaulting` line is required to give a default year value of `2000` so that an exception is not thrown when trying to parse a `LocalDate`.

Reply:
Oh, I see. It is not related to parsing actually.